src/main/java/org/broadinstitute/hellbender/tools/variantdb/nextgen/ExtractFeaturesEngine.java

```diff
@@ -129,18 +136,62 @@ public void traverse() {
                                                                                              INDEL_QUAL_THRESHOLD);
 
         final String userDefinedFunctions = ExtractFeaturesBQ.getVQSRFeatureExtractUserDefinedFunctionsString();
+        Map<String, String> labelForQuery = createQueryLabels(queryLabels);
 
         final StorageAPIAvroReader storageAPIAvroReader = BigQueryUtils.executeQueryWithStorageAPI(
                 featureQueryString,
                 SchemaUtils.FEATURE_EXTRACT_FIELDS,
                 projectID,
                 userDefinedFunctions,
                 useBatchQueries,
-                null);
+                labelForQuery);
 
         createVQSRInputFromTableResult(storageAPIAvroReader);
     }
 
+    private Map<String, String>  createQueryLabels(List<String> labelStringList) {
+        // hardcoded labels are added to the query to make tracking this workflow easier downstream
+        Map<String, String> labelForQuery = new HashMap<String, String>();
+        labelForQuery.put("gvs_tool_name", "extract-features");
+        labelForQuery.put("gvs_query_name", "extract-features");
+        // add additional key value pair labels
+
+        // Each resource can have multiple labels, up to a maximum of 64. -- labelKeys has to be !>64
```

Comment:
technically !> 64 minus your two static labels above right?

Reply:
right---which is why I need to throw if it's more than 62!